### PR TITLE
Add Let's Go! Cup

### PR DESCRIPTION
**Pull request by LegoFigure11**

### CODE REVIEW

File: config/formats.js [hunk 1]
```diff
@@ -206,6 +206,28 @@ let Formats = [
...
+
+		mod: 'gen7',
+		forcedLevel: 50,
+		teamLength: {
+			validate: [1, 1],
+			battle: 1,
+		},
+		ruleset: ['Pokemon', 'Cancel Mod'],
+		banlist: ['Illegal', 'Unreleased'],
+		onValidateTeam: function (team) {
```
Comment (KrisXV) [inline config/formats.js]: This can just be `onValidateSet: function (set, format)`

File: config/formats.js [hunk 2]
```diff
@@ -206,6 +206,28 @@ let Formats = [
...
+			const valid = ['Pikachu', 'Eevee'];
+			let n = 0;
+			for (const set of team) {
+				let baseSpecies = this.getTemplate(set.species).baseSpecies;
+				if (valid.includes(baseSpecies)) n++;
+				if (n !== 1) return ["Only Pikachu and Eevee are allowed in this format!"];
+			}
```
Comment (KrisXV) [inline config/formats.js]: this should just be
```javascript
if (!['eevee', 'pikachu'].includes(toId(set.species))) return [`Only Pikachu and Eevee are allowed in ${format.name}.`];```

File: config/formats.js [hunk 3]
```diff
@@ -206,6 +206,28 @@ let Formats = [
 			'Arceus', 'Reshiram', 'Zekrom', 'Kyurem', 'Xerneas', 'Yveltal', 'Solgaleo', 'Lunala', 'Necrozma-Dusk-Mane', 'Necrozma-Dawn-Wings',
 		],
 	},
+	{
+		name: "[Gen 7] Let's Go! Cup",
+		threads: [`&bullet; <a href="https://www.smogon.com/forums/threads/3639342/">Let&apos;s Go! Cup</a>`],
```
Comment (KrisXV) [inline config/formats.js]: There's no reason to escape the apostrophe, it should just be an ASCII apostrophe.

File: config/formats.js (hunk 1 above)
Comment (KrisXV) [reply config/formats.js]: oh, I see why. There can only be one Pokemon per team. You should just have a check that says ``if (team.length !== 1) return [`Only one Pokemon per team is allowed.`];``

File: config/formats.js [hunk 4]
```diff
@@ -206,6 +206,28 @@ let Formats = [
...
+
+		mod: 'gen7',
+		forcedLevel: 50,
+		teamLength: {
```
Comment (KrisXV) [inline config/formats.js]: adding my team length check mitigates the need for this object

File: config/formats.js [hunk 5]
```diff
@@ -206,6 +206,28 @@ let Formats = [
 			'Arceus', 'Reshiram', 'Zekrom', 'Kyurem', 'Xerneas', 'Yveltal', 'Solgaleo', 'Lunala', 'Necrozma-Dusk-Mane', 'Necrozma-Dawn-Wings',
 		],
 	},
+	{
+		name: "[Gen 7] Let's Go! Cup",
```
Comment (KrisXV) [inline config/formats.js]: I would add a ``desc: `1-Pokemon teams with only Pikachu or Eevee are allowed.`,``

**Comment by LegoFigure11** [comment]
Updated.
Couldn't seem to make this work:
``if (!['eevee', 'pikachu'].includes(toId(set.species))) return [`Only Pikachu and Eevee are allowed in ${format.name}.`];``
so replaced it with this instead.
``if (!['eevee', 'pikachu'].includes(toId(team[0].species))) return [`Only Pikachu and Eevee are allowed in ${format.name}.`];``

**Comment by KrisXV** [comment]
Well it wouldn’t work because you weren’t iterating through the team, but this is also fine

**Pull request merged**